File: 27-remove-element/remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        k = 0

        for i in range(len(nums)):
            if nums[i] != val:
                #partition (quick sort/ quick select)  
                nums[k] = nums[i]
                k += 1
        return k

        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        

File: 27-remove-element/remove-element.java
//when removing the elements that stayed are at the beginning and the removed values are at the end
class Solution {
    public int removeElement(int[] nums, int val) {
        int index = 0;
        for (int i = 0; i< nums.length; i++) {
            if (nums[i] != val) {
                nums[index] = nums[i];
                index++;
            }
        }
        return index;
    }
}